### PR TITLE
DAOS-2280 dfuse: Add daos calls to interception library.

### PR DESCRIPTION
**Pull request by ashleypittman**

### CODE REVIEW

File: src/client/dfuse/il/int_posix.c [hunk 1]
```diff
@@ -209,43 +202,87 @@ ioil_fini(void)
...
-			     gah_info.dfuse_id);
-		return false;
-	}
-
-	entry->common.projection = &projections[gah_info.cli_fs_id];
 	entry->pos = 0;
 	entry->flags = flags;
 	entry->status = DFUSE_IO_BYPASS;
+
+	svcl = daos_rank_list_parse("0", ":");
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfuse/il/int_posix.c]: Agreed, but how should the IL get this data?  I suppose it needs to be passed out in the ioctl?

File: src/client/dfuse/dfuse_ioctl.h [hunk 2]
```diff
@@ -24,17 +24,17 @@
...
 #define DFUSE_IOCTL_GAH_NUMBER 0xC1 /* Number of the GAH IOCTL.  Also arbitrary */
 #define DFUSE_IOCTL_VERSION 4       /* Version of ioctl protocol */
 
 struct dfuse_gah_info {
 	int version;
-	struct ios_gah gah;
-	int dfuse_id;
-	int cli_fs_id;
+	daos_obj_id_t	oid;
+	const char pool[NAME_MAX];
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfuse/dfuse_ioctl.h]: Now I've got a better understanding of this I suspect you're right.  I'll change the name of this struct at the same time.

File: src/client/dfuse/il/int_posix.c [hunk 3]
```diff
@@ -347,9 +384,21 @@ dfuse_close(int fd)
 		goto do_real_close;
 
 	DFUSE_LOG_INFO("close(fd=%d." GAH_PRINT_STR ") intercepted, bypass=%s",
-		       fd, GAH_PRINT_VAL(entry->common.gah),
+		       fd, GAH_PRINT_VAL(entry),
 		       bypass_status[entry->status]);
 
+#if 0
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfuse/il/int_posix.c]: Yes.  This is the problem with dup(), if we close them then we can no longer access other the same file under a different fd.  I'm working on fixing this.

File: src/client/dfuse/il/int_posix.c [hunk 4]
```diff
@@ -168,90 +180,161 @@ ioil_init(void)
...
 ioil_fini(void)
 {
-	if (ioil_initialized) {
-		free(projections);
-	}
-	ioil_initialized = false;
+	ioil_ioc.ioc_initialized = false;
 
-	__sync_synchronize();
+	if (ioil_ioc.ioc_open_fd_count > 0) {
```
Comment (mchaarawi) [inline src/client/dfuse/il/int_posix.c]: i understand from this that we only support single pool and container currently?

**Pull request merged**